libbpf-rs/src/query.rs

```diff
@@ -419,14 +421,26 @@ impl Iterator for ProgInfoIter {
                 return None;
             }
 
-            let prog = ProgramInfo::load_from_fd(fd, &self.opts);
-            let _ = close(fd);
+            return Some(unsafe { OwnedFd::from_raw_fd(fd) });
+        }
+    }
+}
 
-            match prog {
-                Ok(p) => return Some(p),
-                // TODO: We should consider bubbling up errors properly.
-                Err(_err) => (),
-            }
+impl Iterator for ProgInfoIter {
+    type Item = ProgramInfo;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let fd = match self.next_valid_fd() {
+            Some(fd) => fd,
+            None => return None,
+        };
```

Comment:
Should be the same as:
```suggestion
        let fd = self.next_valid_fd()?;
```
Right?